lib/hl7/message.ex

```diff
@@ -509,26 +509,50 @@ defmodule HL7.Message do
   end
 
   defp validate_text(raw_text, options) do
-    encoded_text =
-      if options[:accept_latin1] == true do
-        :unicode.characters_to_binary(raw_text, :latin1)
+    if options[:accept_latin1] do
```

Comment:
I've adjusted this such that it can possibly bail early, but it will only do so in case validate is true and accept_latin1 is false, and since we only transcode is accept_latin1 is true, it doesn't feel right (i.e., there's an execution path we'll never go down). 

Perhaps we need a case where accept_latin1 is true, we transcode but run into an error doing so, however I can not think of a case :)